src/parsers/variable.cr

```diff
@@ -10,11 +10,14 @@ module Mint
 
       raise error unless value
 
-      Ast::Variable.new(
+      node = Ast::Variable.new(
         from: start_position,
         value: value,
         to: position,
         input: data)
+
+      self << node if track
```

Comment:
In some cases it's not necessary to track a variable because the source range is covered by the parent. 

For example when hovering over a functions name it should not resolve the name variable but the function.